src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -1920,13 +1872,17 @@ int D4NFilterObject::D4NFilterDeleteOp::delete_obj(const DoutPrefixProvider* dpp
       }
       std::vector<std::string> responses;
       ret = blockDir->exec(dpp, responses, y);
-      if (ret < 0) {
+      if (ret < 0 && ret != -ENOENT) {
```

Comment:
also please do not make additions to the delete_obj code here without discussing with me.

Reply:
I don't recall explicitly changing your `delete_obj` additions like this so this might've been a rebase error. I would otherwise make sure to run any alterations by you first, so this definitely wasn't intentional. Apologies!